eng/common/testproxy/onboarding/common-asset-functions.ps1

```diff
@@ -0,0 +1,273 @@
+      $this.Year = [int]$Matches["year"]
+      $this.Month = [int]$Matches["month"]
+      $this.Day = [int]$Matches["day"]
+      $this.Revision = [int]$Matches["revision"]
+    }
+    else {
+      # This should be a Write-Error however powershell apparently cannot utilize that
+      # in the constructor in certain cases
+      Write-Warning "Version String '$($VersionString)' is invalid and cannot be parsed"
+      exit 1
+    }
+  }
+  [bool] IsGreaterEqual([string]$OtherVersionString) {
+    [Version]$OtherVersion = [Version]::new($OtherVersionString)
+    if ($this.Year -lt $OtherVersion.Year) {
+      return $false
+    }
+    elseif ($this.Year -eq $OtherVersion.Year) {
+      if ($this.Month -lt $OtherVersion.Month) {
+        return $false
+      }
+      elseif ($this.Month -eq $OtherVersion.Month) {
+        if ($this.Day -lt $OtherVersion.Day) {
+          return $false
+        }
+        elseif ($this.Day -eq $OtherVersion.Day) {
+          if ($this.Revision -lt $OtherVersion.Revision) {
+            return $false
+          }
+        }
+      }
+    }
+    return $true
+  }
+}
+
+Function Test-Exe-In-Path {
+  Param([string] $ExeToLookFor, [bool]$ExitOnError = $true)
+  if ($null -eq (Get-Command $ExeToLookFor -ErrorAction SilentlyContinue)) {
+    if ($ExitOnError) {
+      Write-Error "Unable to find $ExeToLookFor in your PATH"
+      exit 1
+    }
+    else {
+      return $false
+    }
+  }
+
+  return $true
+}
+
+Function Test-TestProxyVersion {
+  param(
+    [string] $TestProxyExe
+  )
+
+  Write-Host "$TestProxyExe --version"
+  [string] $output = & "$TestProxyExe" --version
+
+  [Version]$CurrentProxyVersion = [Version]::new($output)
+  if (!$CurrentProxyVersion.IsGreaterEqual($MinTestProxyVersion)) {
+    Write-Error "$TestProxyExe version, $output, is less than the minimum version $MinTestProxyVersion"
+    Write-Error "Please refer to https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md#installation to upgrade your $TestProxyExe"
+    exit 1
+  }
+}
+
+Function Get-Repo-Language {
+
+  $GitRepoOnDiskErr = "This script can only be called from within an azure-sdk-for-<lang> repository on disk."
+  # Git remote -v is going to give us output similar to the following
+  # origin  git@github.com:Azure/azure-sdk-for-java.git (fetch)
+  # origin  git@github.com:Azure/azure-sdk-for-java.git (push)
+  # upstream        git@github.com:Azure/azure-sdk-for-java (fetch)
+  # upstream        git@github.com:Azure/azure-sdk-for-java (push)
+  # We're really only trying to get the language from the git remote
+  Write-Host "git remote -v"
+  [array] $remotes = & git remote -v
+  foreach ($line in $remotes) {
+    Write-Host "$line"
+  }
+
+  # Git remote -v returned "fatal: not a git repository (or any of the parent directories): .git"
+  # and the list of remotes will be null
+  if (-not $remotes) {
```

Comment:
Powershell doesn't store the error into the var?

Reply:
Nah that goes to stderr IIRC.

Reply:
![image](https://github.com/Azure/azure-sdk-tools/assets/45376673/4d73ae76-ecff-40e6-8a68-36a3cb1dcba1)

Confirmed.